Lab4/src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,62 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RectangularMap implements IWorldMap {
+
+    private int width;
+    private int height;
+    private List<Animal> listOfAnimals;
+
+    public RectangularMap(int width, int height) {
+        this.width = width;
+        this.height = height;
+        listOfAnimals = new ArrayList<Animal>();
+    }
+
+    public List<Animal> getListOfAnimals() {
+        return listOfAnimals;
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+
+        for (Animal animal : listOfAnimals) {
+            if (animal.isAt(position) || !position.precedes(new Vector2d(this.width, this.height)) || !position.follows(new Vector2d(0,0))) {
```

Comment:
sprawdzanie brzegów dla każdego zwierzątka w pętli jest niepotrzebne, można to zrobić raz. W pętli wystarczy sprawdzac konflikty z zwierzętami, a to z kolei robi już metoda `isOccupied()` więc można ją tu tez wykorzystać.